generate_chip.py

```diff
@@ -581,9 +585,75 @@ def add_footprint_variant(
         else:
             raise ValueError('Either gap or footprints must be set')
 
+        # Generate 3D models (for certain package types)
+        if package_type in ['RESC', 'CAPC']:
+            uuid_3d = uuid('pkg', full_name, '3d')
+            if generate_3d_models:
+                generate_3d(library, package_type, full_name, uuid_pkg, uuid_3d, config)
+            package.add_3d_model(Package3DModel(uuid_3d, Name(full_name)))
+            for footprint in package.footprints:
+                footprint.add_3d_model(Footprint3DModel(uuid_3d))
+
         package.serialize(path.join('out', library, category))
 
 
+def generate_3d(
+    library: str,
+    package_type: str,
+    full_name: str,
+    uuid_pkg: str,
+    uuid_3d: str,
+    config: ChipConfig,
+) -> None:
+    import cadquery as cq
+    from cadquery_helpers import StepAssembly
+
+    print(f'Generating pkg 3D model "{full_name}": {uuid_3d}')
+
+    length = config.body.length
+    width = config.body.width
+    height = config.body.height
+
+    fillet = min(width, height) * 0.05
+    edge = min(length * 0.15, 0.4)
+    translation = (0, 0, height / 2)
+    edge_offset = length / 2 - edge
+
+    inner = cq.Workplane("XY") \
+        .box(length - 2 * edge, width, height) \
+        .edges('+X').fillet(fillet) \
+        .translate(translation)
+    left = cq.Workplane("XY") \
+        .box(edge, width, height) \
+        .edges('+X or <X').fillet(fillet) \
+        .translate(translation) \
+        .translate((-edge_offset - edge / 2, 0, 0))
+    right = cq.Workplane("XY") \
+        .box(edge, width, height) \
+        .edges('+X or >X').fillet(fillet) \
+        .translate(translation) \
+        .translate((edge_offset + edge / 2, 0, 0))
+
+    if package_type == 'RESC':
+        inner_color = cq.Color('gray16')
+    elif package_type == 'CAPC':
+        inner_color = cq.Color('bisque3')
+    else:
+        raise RuntimeError(f'Unsupported 3D package type: {package_type}')
+
+    assembly = StepAssembly(full_name)
+    assembly.add_body(inner, 'inner', inner_color)
+    assembly.add_body(left, 'left', cq.Color("gainsboro"))
+    assembly.add_body(right, 'right', cq.Color("gainsboro"))
+
+    dir_path = path.join('out', library, 'pkg', uuid_pkg)
+    if not (path.exists(dir_path) and path.isdir(dir_path)):
+        makedirs(dir_path)
```

Comment:
Maybe move `makedirs()` into `StepAssembly.save()` so not every script has to do that?